sys/kern/vfs_aio.c

```diff
@@ -1470,27 +1449,31 @@ aiocb_save_aiocb(struct kaiocb *kjob, const void *ujobptrp)
 	if (ujobptrp == NULL)
 		memset(&kjob->ujobptr, 0, sizeof(kjob->ujobptr));
 	else
-		memcpy(&kjob->ujobptr, ujobptrp, sizeof(long));
+		memcpy(&kjob->ujobptr, ujobptrp, sizeof(intcap_t));
 }
 
 static int
-aiocb_store_aiocb(kaiocb_t ** __capability ujobp, struct kaiocb *kjob)
+aiocb_store_aiocb(struct aiocb ** __capability ujobp, struct kaiocb *kjob)
 {
-	long ujob;
+	intcap_t ujob;
 
 	if (kjob == NULL)
 		ujob = 0;
 	else
-		memcpy(&ujob, &kjob->ujobptr, sizeof(long));
+		memcpy(&ujob, &kjob->ujobptr, sizeof(intcap_t));
 
+#if __has_feature(capabilities)
+	return (copyoutcap(&ujob, ujobp, sizeof(ujob)) == 0 ? 0 : -1);
```

Comment:
Does it make sense to just always do the copyoutcap?

Reply:
Hmm, well, I think what I would really like would be to have 'sucap' be always available and have a fallback that maps to suword for the non-cap case.